### ISSUE TITLE
Renovate updates a dependency only once

### ISSUE THREAD

**Issue by osigida** [renovatebot/renovate#12840]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v29.23.3

### Please select which platform you are using if self-hosting.

Bitbucket Cloud (bitbucket.org)

### Describe the bug

When renovate has updated a dependency one, it ignores it for the second time.

I have a multi-module gradle project. 
Renovate has updated org.slf4j:slf4j-api to v1.7.32. (merged successfully) 
Then a developer introduced same dependency v1.7.31 in a new spot.

Renovate skips the update because it detects already closed PR 

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Setting current branch to master (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
DEBUG: latest commit (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
       "branchName": "master",
       "latestCommitDate": "2021-11-24T14:51:43+01:00"
DEBUG: getBranchPr(renovate/org.slf4j-slf4j-api-1.7.x) (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
DEBUG: findPr(renovate/org.slf4j-slf4j-api-1.7.x, undefined, open) (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
DEBUG: getPrList() (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
DEBUG: branchExists=true (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
DEBUG: dependencyDashboardCheck=undefined (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
DEBUG: recreateClosed is false (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
DEBUG: findPr(renovate/org.slf4j-slf4j-api-1.7.x, Update dependency org.slf4j:slf4j-api to v1.7.32, !open) (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
DEBUG: getPrList() (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
DEBUG: Found PR #127 (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
DEBUG: Found closed PR with current title (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
DEBUG: Closed PR already exists. Skipping branch. (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
       "prTitle": "Update dependency org.slf4j:slf4j-api to v1.7.32"
DEBUG: Merged PR is blocking this branch (repository=repository_name_here, branch=renovate/org.slf4j-slf4j-api-1.7.x)
       "pr": 127
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
This is currently by design, but I think we can loosen it so that an existing *merged* PR does not block the creation of a new PR, it only blocks *automerge* the second time, if it was enabled.

**Comment by osigida** [comment]
I guess the status of the last PR should be considered: 
if the PR was merged - free to open PR again
if the PR was rejected/declined - the version should be skipped

**Comment by viceice** [comment]
> I guess the status of the last PR should be considered: if the PR was merged - free to open PR again if the PR was rejected/declined - the version should be skipped

This would also cause a new PR if the previous was reverted after merge. 🤔

**Comment by osigida** [comment]
@viceice you still have and ability to decline version update, and renovate will notice that next time, right?

**Comment by viceice** [comment]
Sure, but what if i don't remeber that i've reverted it 😉 

Disable automerge wourld be mandatory for that, otherwise we can get some kind of ping-pong

**Comment by rarkins** [comment]
Maybe we have the ability to add a prBodyNotes entry saying that it could possibly be a duplicate PR from previously
